testsuite/tests/gc-roots/globroot.ml

```diff
@@ -0,0 +1,77 @@
+(* 
+   modules = "globrootsprim.c"
+*)
+
+module type GLOBREF = sig
+  type t
+  val register: string -> t
+  val get: t -> string
+  val set: t -> string -> unit
+  val remove: t -> unit
+end
+
+module Classic : GLOBREF = struct
+  type t
+  external register: string -> t = "gb_classic_register"
+  external get: t -> string = "gb_get"
+  external set: t -> string -> unit = "gb_classic_set"
+  external remove: t -> unit = "gb_classic_remove"
+end
+
+module Generational : GLOBREF = struct
+  type t
+  external register: string -> t = "gb_generational_register"
+  external get: t -> string = "gb_get"
+  external set: t -> string -> unit = "gb_generational_set"
+  external remove: t -> unit = "gb_generational_remove"
+end
+
+module Test(G: GLOBREF) = struct
+
+  let size = 1024
+
+  let vals = Array.init size Int.to_string
+
+  let a = Array.init size (fun i -> G.register (Int.to_string i))
+
+  let check () =
```

Comment:
This function doesn't actually seem to get called. Not sure that's intended, it feels like it should be called after each `change ()` call in `test`

Reply:
That's something I have been wondering myself. The original test in trunk (https://github.com/ocaml/ocaml/blob/trunk/testsuite/tests/gc-roots/globroots.ml) too doesn't call `check` function. But I think you're right, `check` is now called after `change`.